xdsl/dialects/memref.py

```diff
@@ -61,8 +61,18 @@
 _MemRefTypeElement = TypeVar("_MemRefTypeElement", bound=Attribute)
 
 
+class ShapeType(ABC):
```

Comment:
Not sure this should really live here. I would actually argue this fits better into builtin than into memref. What do others think?

Reply:
Not sure either, it reads wrong to inherit from memref.ShapeType indeed

I'd be happy with built-in, but maybe this should be a new file on its own?

Reply:
I'm happy with either, it's a good idea

Reply:
Yeah, I guess this could be something "attribute_prototypes" or whatever. But until we have several of those and a need to move them, I would just throw it into builtin.

Reply:
I would put this in builtin to be honest. That's how this is done in MLIR at least (since it is used by `Vector` and `Tensor` and `Memref`).